spimdisasm/mips/symbols/MipsSymbolBase.py

```diff
@@ -193,36 +193,32 @@ def shouldMigrate(self) -> bool:
         return False
 
 
-    def renameBasedOnType(self):
-        if not common.GlobalConfig.AUTOGENERATED_NAMES_BASED_ON_DATA_TYPE:
-            return
-
+    def autogenerateName(self, contextSym, inSymbolOffset):
         if not self.contextSym.isAutogenerated:
             return
 
-        if self.isFloat(0):
-            self.contextSym.name = f"FLT_{self.vram:08X}"
-        elif self.isDouble(0):
-            self.contextSym.name = f"DBL_{self.vram:08X}"
-        elif self.isString():
-            self.contextSym.name = f"STR_{self.vram:08X}"
-        elif self.isPascalString():
-            self.contextSym.name = f"PSTR_{self.vram:08X}"
-
+        if common.GlobalConfig.AUTOGENERATED_NAMES_BASED_ON_FILE_NAME and self.parent is not None:
+            offset = self.inFileOffset + inSymbolOffset
+            contextSym.name = f"{self.parent.name}{self.sectionType.toStr()}_{offset:06X}"
+        elif common.GlobalConfig.AUTOGENERATED_NAMES_BASED_ON_DATA_TYPE:
+            if self.isFloat(0):
+                contextSym.name = f"FLT_{self.vram:08X}"
+            elif self.isDouble(0):
+                contextSym.name = f"DBL_{self.vram:08X}"
+            elif self.isString():
+                contextSym.name = f"STR_{self.vram:08X}"
+            elif self.isPascalString():
+                contextSym.name = f"PSTR_{self.vram:08X}"
 
     def analyze(self):
-        self.renameBasedOnType()
+        self.autogenerateName(self.contextSym, 0)
 
-        byteStep = 4
-        if self.contextSym.isByte():
-            byteStep = 1
-        elif self.contextSym.isShort():
-            byteStep = 2
```

Comment:
I wonder if this has any measurable impact on runtime since this may change analyzis from each 4 bytes to every byte

Reply:
Looks like yes:

develop: 27.61 secs
this branch (without --name-vars-by-file): 30.07 secs

Should I restore the old behavior but add a check for `AUTOGENERATED_NAMES_BASED_ON_FILE_NAME` here?

Reply:
Many in-between symbols are getting properly named with this change, seems like scanning every byte is the way to go.

So no, even if runtime is a bit slower this should be the new behavior